mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go

```diff
@@ -57,7 +63,14 @@ func resourceLoggingProjectSinkCreate(d *schema.ResourceData, meta interface{})
 	id, sink := expandResourceLoggingSink(d, "projects", project)
 	uniqueWriterIdentity := d.Get("unique_writer_identity").(bool)
 
-	_, err = config.NewLoggingClient(userAgent).Projects.Sinks.Create(id.parent(), sink).UniqueWriterIdentity(uniqueWriterIdentity).Do()
+	customWriterIdentity := d.Get("custom_writer_identity").(string)
+
+	if (customWriterIdentity != ""){
```

Comment:
rather than this if statement, what would you think about something like:

```go
client := config.NewLoggingClient(userAgent).Projects.Sinks.Create(id.parent(), sink)
if (customWriterIdentity != "") {
  client = client.UniqueWriterIdentity(uniqueWriterIdentity)
}

_, err := client.Do()
```

not sure if this would work or if `client` is the right variable name - just wondering if we can isolate the part that's actually changing based on the writer identity.

Reply:
updated the code based on the review suggestion.